pegasus/sites.v3/code.org/views/admins_email_signup_form.html

```diff
@@ -0,0 +1,48 @@
+<script src="https://www.google.com/recaptcha/api.js"></script>
+<script>
+ function timestamp() { var response = document.getElementById("g-recaptcha-response"); if (response == null || response.value.trim() == "") {var elems = JSON.parse(document.getElementsByName("captcha_settings")[0].value);elems["ts"] = JSON.stringify(new Date().getTime());document.getElementsByName("captcha_settings")[0].value = JSON.stringify(elems); } } setInterval(timestamp, 500);
+</script>
```

Comment:
our best practice is to avoid adding js directly in html files (or haml, or erb), and instead to extract it into its own js file. this allows us to catch errors via the linter, and also to ensure that it is properly polyfilled to work across browsers. after launch, we should follow up to find a way to get this js out of html / haml if possible.

I see this is copied from salesforce-form-test.haml . did you have to write this code snippet, or is it copied from the recaptcha docs?

Reply:
That's good to know! Should I move the js now before launch? If yes, can you tell me where it's best to put the file?

And the form code came directly from Salesforce (I just added the `required` attribute on the inputs). We made the test page so we could make sure it was sending when not on localhost to make sure the recaptcha was showing up. It was giving an "invalid domain key" error on the dev environment.

Reply:
> That's good to know! Should I move the js now before launch? If yes, can you tell me where it's best to put the file?

if there is time before launch that would be great! can you point me toward the docs you were following to get captcha working?

Reply:
per offline convo, this js was provided by salesforce. @kelbyhawn it would be great if you could please try to extract a js file. our system follows a naming convention, and based on that, the file would go in apps/src/sites/code.org/pages/views/admins_email_signup_form.js, and you would need to add an "entry point" for it here: https://github.com/code-dot-org/code-dot-org/blob/5441b5f118cf14428c9953dddeb93b766128b415/apps/Gruntfile.js#L783-L784 

to get this js to work properly in that file, I think you will want to wrap the extracted js in a call to 
```js
$(document).ready(function() {
  ...
});
```
no worries if you run out of time before launch. wherever the js (and html) end up, please note that they were provided by salesforce.

Reply:
also you will need to run `cd apps; yarn build` to get the new js file to show up. (apologies, not sure how much of this you already know)

Reply:
In this particular case, it looks like salesforce has provided a well-tested js snippet that will work on all browsers (I am guessing that because it uses es5 `var` instead of es6 `let` or `const`). so, the danger of NOT moving this code to apps/src is somewhat lower, as long as it is well-commented that this code came vetted by salesforce.

Reply:
if this is your first time adding something to apps/src/sites, then it may be best for us to pair on this and also just make sure this convention is going to work for you

Reply:
Closing the loop: going to merge this PR as-is and create a followup PR to test the updated JS implementation on the https://staging.code.org/salesforce-form-test page before implementing here in another followup PR.